v2/contribute/decisions/thirdparty/0007-check-for-enabled-recipes.mdx

```diff
@@ -21,8 +21,14 @@ Should each of the recipe check if the recipe is enabled in the core?
 
 ## Considered Options
 
-* Recipes should not check
+* Recipe should not check
+  * API Should not check
+  * Functions should not check
 * Recipes should check
+  * API should check
+  * Functions should not check
+
+Reason why we don't check in the functions because it allows the user to use the functions even if that recipe is not enabled for that tenant, and is not exposed to pose security risk. However, API check would need to be considered as it is publicly exposed.
```

Comment:
```suggestion
Reason why we don't check in the functions is because it allows the user to use the functions even if that recipe is not enabled for that tenant. This makes the dev experience is better. Furthermore, it does not pose any security risk since these recipe functions are not exposed publicly. However in the case of APIs, we need to check if the recipe is enabled for the tenant because they are publicly exposed and want to avoid a situation in which a tenant can sign up, for example, with emailpassword even though that recipe is not enabled for them in the core.
```